plugins/prometheus_plugin/metrics.hpp

```diff
@@ -34,6 +38,19 @@ struct catalog_type {
 
    Gauge& num_peers;
    Gauge& num_clients;
+   std::vector<std::string> addresses;
+   std::vector<std::reference_wrapper<Gauge>> accepting_blocks_gauges;
+   std::vector<std::reference_wrapper<Gauge>> last_received_blocks_gauges;
+   std::vector<std::reference_wrapper<Gauge>> first_available_blocks_gauges;
+   std::vector<std::reference_wrapper<Gauge>> last_available_blocks_gauges;
+   std::vector<std::reference_wrapper<Gauge>> unique_first_block_counts_gauges;
+   std::vector<std::reference_wrapper<Gauge>> latency_gauges;
+   std::vector<std::reference_wrapper<Gauge>> bytes_received_gauges;
+   std::vector<std::reference_wrapper<Gauge>> last_bytes_received_gauges;
+   std::vector<std::reference_wrapper<Gauge>> bytes_sent_gauges;
+   std::vector<std::reference_wrapper<Gauge>> last_bytes_sent_gauges;
+   std::vector<std::reference_wrapper<Gauge>> connection_start_time_gauges;
+   std::vector<std::reference_wrapper<Gauge>> log_p2p_address_gauges;
```

Comment:
Seems like a vector of struct would be better than all these parallel vectors.

Reply:
Done.

Reply:
Removed entirely since they turn out to be unnecessary.